mission/Chapter04/Mission2&3/src/hooks/useLocalStorage.ts

```
@@ -0,0 +1,29 @@
+export const useLocalStorage = (key: string) => {
+  const setItem = (value: unknown) => {
+    try {
+      window.localStorage.setItem(key, JSON.stringify(value));
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const getItem = () => {
+    try {
+      const item = window.localStorage.getItem(key);
+
+      return item ? JSON.parse(item) : null;
```

JAON.parse 는 파싱 실패 시 콘솔에 에러가 계속 찍히므로, 에러를 콘솔에 찍는 것보다 silent fail 혹은 fallback value를 주는 게 사용자 경험에 더 좋을 것 같습니당 !!